### PR TITLE
5pm-b Backend Schedule Operations

### LINKED ISSUE #50: Create Entity and Repository for Schedules, and add some fake data (similar to how that works in the ucsb-cs-las app
In this story, we just add to the backend, the database table for Schedules, and we add a data.sql file that populates that table with some temporary data.

Once that is done, the stories to list that data, add to that table, edit existing entries and delete entries can all be separate stories.

# User Story

As a student I can create and store named schedules so that I can plan my classes.

# Acceptance Criteria

- [x] There is a way to store schedules in a database
- [x] There is a way to store schedule items in a database
- [x] There is a repository to interact with the MongoDB database

# Implementation Todos

Back end:

- [x] Getter and setter methods for the Schedule object.
- [x] Getter and setter methods for the ScheduleItem object.

...

Testing:

- [x] Test that schedule objects can be created
- [x] Test that schedule item objects can be created
- [x] Test getter and setter methods on schedule objects
- [x] Test getter and setter methods on schedule item objects

### PR DESCRIPTION
**Pull request by Mitchell57**
This pull request adds entities for Schedule and ScheduleItem, as well as repositories, controllers, and tests for all new classes.

Fixes #50 and #35

### CODE REVIEW

**Comment by pconrad** [comment]
Please address merge conflicts, and get a team member's code review before staff review this

**Comment by pconrad** [comment]
Please also remove `.DS_Store` files from this branch, and add `.DS_Store` to `.gitignore` before we review it.

**Comment by Mitchell57** [comment]
The only things not being tested are the onSuccess and onError declarations in the AddSchedForm and ScheduleSearchForm, but I'm not sure how to trigger those from the Schedule page

**Review by seanaocostello — COMMENTED** [review]
Looks good to me!

**Review by andyyyho — DISMISSED** [review]
Looks good

**Review by seanaocostello — APPROVED** [review]

**Review by seanaocostello — DISMISSED** [review]
Still looks good!

**Review by pconrad — CHANGES_REQUESTED** [review]
This PR deletes `javascript/.env.local.SAMPLE`, but that is a critical file.

That files needs to be restored before this is mergable.

File: docs/heroku.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ This consists of the following steps:
 
 ```bash
 cp javascript/.env.local javascript/.env.production
-cp secrets-heroku.properties.SAMPLE secrets-heroku.properties
+cp secrets-heroku.properties secrets-heroku.properties
```
Comment (pconrad) [inline docs/heroku.md:13]: This change is incorrect.     Please revert this change to the original

File: javascript/package-lock 2.json [hunk 2]
```diff
@@ -0,0 +1,14713 @@
+{
```
Comment (pconrad) [inline javascript/package-lock 2.json:1]: This file should  not be checked into the code base.  Not sure why it exists.  Please remove

File: javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js [hunk 3]
```diff
@@ -1,8 +1,60 @@
 import React from "react";
+import { Button } from "react-bootstrap";
 import BootstrapTable from 'react-bootstrap-table-next';
 import { reformatJSON } from 'main/utils/BasicCourseTableHelpers';
 
+
+
+//const buttonFormatter = (cell, row, rowIndex, formatExtraData) => {
```
Comment (pconrad) [inline javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js:8]: Please remove the commented out code.

In general, commented out code should not be committed to main. Commenting out code is a strategy used during the development of code, but it should not end up in the final product.

**Review by pconrad — CHANGES_REQUESTED** [review]
secrets-heroku.properties.SAMPLE should not have been deleted

secrets-localhost.properties.SAMPLE should not have been deleted.

please restore these

**Review by pconrad — CHANGES_REQUESTED** [review]
Please remove all of the .DS_Store files that are part of this commit.

**Comment by mara-downing** [comment]
Closing for now, linked in w21 issue #40 for students to reference.

**Pull request closed without merging**